File: Java/src/main/java/com/example/stack/Stack.java
package com.example.stack;

class StackNode {
    int data;
    StackNode next;
    public StackNode(int data) {
        this.data = data;
        this.next = null;
    }
}

public class Stack {
    StackNode head;
    int length;
    public void push(int data) {
        StackNode node = new StackNode(data);
        node.next = head;
        head = node;
        length++;
    }

    public int pop() {
        if (head == null) {
            return -1;
        }
        int data = head.data;
        head = head.next;
        length--;
        return data;
    }

    public int top() {
        if (head == null) {
            return -1;
        }
        return head.data;
    }

    public int size() {
        return length;
    }

    public boolean isEmpty() {
        return length == 0;
    }
}

File: dart/src/stack/Stack.dart
class StackNode {
  int? data;
  StackNode? next;
  StackNode(this.data) {}
}

class Stack {
  StackNode? head;
  int length = 0;
  void push(int data) {
    var newNode = StackNode(data);
    newNode.next = this.head;
    this.head = newNode;
    this.length = this.length + 1;
  }

  int? pop() {
    if (this.head == null) {
      return null;
    }
    var data = this.head?.data;
    this.head = this.head?.next;
    this.length = this.length - 1;
    return data;
  }

  int? top() {
    return this.head?.data;
  }

  int size() {
    return this.length;
  }

  bool isEmpty() {
    return this.head == null;
  }
}
